pkg/otp/otp.go

```diff
@@ -45,47 +45,43 @@ func Calculate(name string, sec gopass.Secret) (twofactor.OTP, string, error) {
 	}
 
 	if strings.HasPrefix(secKey, "otpauth://") {
-		return twofactor.FromURL(secKey) //nolint:wrapcheck
+		debug.Log("parsing otpauth:// URL %q", out.Secret(secKey))
+
+		k, err := otp.NewKeyFromURL(secKey)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse otpauth URL: %w", err)
+		}
+
+		return k, nil
 	}
 
-	otp, err := twofactor.NewGoogleTOTP(twofactor.Pad(secKey))
+	debug.Log("assembling otpauth URL from secret only (%q), using defaults", out.Secret(secKey))
```

Comment:
This is logging a secret, not great.

Reply:
Yes, it's logging a secret. But it's doing so responsibly. See https://github.com/gopasspw/gopass/blob/master/pkg/debug/debug.go#L210 and https://github.com/gopasspw/gopass/blob/master/internal/out/print.go#L27.